hw8/todolist/src/components/TagButtons/TagButtons.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useState } from "react";
+import { tags, uniqueTagColors } from "../../App";
+import { Link, useLocation } from "react-router-dom";
+
+interface Props {
+  newTag: string;
+  setNewTag: (newTag: string) => void;
+  isLink?: boolean;
+}
+const TagButtons = ({ newTag, setNewTag, isLink = false }: Props) => {
+  const [tagColors, setTagColors] = useState(["#FFF", "#FFF", "#FFF", "#FFF"]);
+
+  const location = useLocation();
+
+  return (
+    <>
+      {tags.map((tag, index) => {
+        let nextTagColors = ["#FFF", "#FFF", "#FFF", "#FFF"];
+        nextTagColors[index] = uniqueTagColors[index];
+        let isSelected = newTag === tag;
+        return isLink ? (
+          <Link
```

Comment:
Also, it's better to extract long component parts to the separated component and ise it like (isLink ? <Component1/> : <Component2/>). It hard to read if there are many lines of the code